src/isl/util.rs

```diff
@@ -465,3 +465,65 @@ pub enum RangeType {
     NonNegativeInteger, // used by byte_length, container_length and codepoint_length to specify non negative integer range
     Any,                // used for any other range types (e.g. Integer, Float, Timestamp, Decimal)
 }
+
+/// Represents annotation modifiers for [annotations] constraint.
+/// Grammar: <ANNOTATIONS_MODIFIER> ::= required::
+///                          | ordered::
+///                          | closed::
+/// [annotations]: https://amzn.github.io/ion-schema/docs/spec.html#annotations
+#[derive(Debug, Clone, PartialEq)]
+pub enum AnnotationModifier {
+    Required,
+    Ordered,
+    Closed,
+    Any, // any other annotation that will be considered as open content
```

Comment:
It is used  to define open content annotation.

Reply:
Yes, but why is it necessary? This doesn't do anything with the open content, and if you see this, you might know there is open content, but you don't know what the open content is because this is `Any` rather than `Any(String)` (or something like that).

Why not just ignore the extra annotations (like ion-schema-kotlin) or prohibit them (like the Ion Schema spec)?

Reply:
So here I am using `From<OwnedElement>` for `AnnotaionModfier` so that I can just call owned_element.into() and it will convert to `AnnotationModifier`. While doing this I would have to represent the open_content with some annotation. That is why Any. This is then filtered to remove all `AnnotaionModifier::Any `from the list of annotations. 

I can change this to be Any(String) but I don't think we want to store this open content when we can just ignore it.

Another solution: I can just filter out open content at first then convert annotations to `AnnotationModifier`?

Reply:
I think the second solution might be better. We don't really want people to be able to construct an `AnnotationConstraint` instance that has a modifier of `Any`.

A third option might be to implement `into()` so that it returns `Option<AnnotationModifier>`. Then you could return `None` for open content.

However, if you think that using boolean fields or a bitmask field instead of a vector of enum values is a good idea, the this concern will become irrelevant, so don't worry about figuring anything out here if you're planning on changing how the modifiers are modeled.

Reply:
Yes, true. I was just going to mention that removing `AnnotationModifier` and moving to boolean fields would resolve this issue.